librz/include/rz_util/rz_buf.h

```diff
@@ -211,6 +211,68 @@ DEFINE_RZ_BUF_WRITE_BLE(16)
 DEFINE_RZ_BUF_WRITE_BLE(32)
 DEFINE_RZ_BUF_WRITE_BLE(64)
 
+#define DEFINE_RZ_BUF_READ_OFFSET_BLE(size) \
+	static inline bool rz_buf_read_ble##size##_offset(RZ_NONNULL RzBuffer *b, RZ_NONNULL RZ_INOUT ut64 *offset, RZ_NONNULL RZ_OUT ut##size *result, bool big_endian) { \
+		rz_return_val_if_fail(b &&offset &&result, false); \
+		if (!rz_buf_read_ble##size##_at(b, *offset, result, big_endian)) { \
+			return false; \
+		} \
+		*offset += sizeof(*result); \
+		return true; \
+	}
+
+#define DEFINE_RZ_BUF_WRITE_OFFSET_BLE(size) \
+	static inline bool rz_buf_write_ble##size##_offset(RZ_NONNULL RzBuffer *b, RZ_NONNULL RZ_INOUT ut64 *offset, ut##size value, bool big_endian) { \
+		rz_return_val_if_fail(b &&offset, false); \
+		if (!rz_buf_write_ble##size##_at(b, *offset, value, big_endian)) { \
+			return false; \
+		} \
+		*offset += sizeof(value); \
+		return true; \
+	}
+
+/**
+ * \brief Read a big endian or little endian (ut16, ut32, ut64) at the specified offset in the buffer and shifts the offset.
+ * \param b ...
+ * \param offset ...
+ * \param result ...
+ * \param big_endian ...
+ * \return Return the status of the operation.
+ */
+DEFINE_RZ_BUF_READ_OFFSET_BLE(16)
+DEFINE_RZ_BUF_READ_OFFSET_BLE(32)
+DEFINE_RZ_BUF_READ_OFFSET_BLE(64)
+
+static inline bool rz_buf_read8_offset(RZ_NONNULL RzBuffer *b, RZ_NONNULL RZ_INOUT ut64 *offset, RZ_NONNULL RZ_OUT ut8 *result) {
+	rz_return_val_if_fail(b && offset && result, false);
+	if (!rz_buf_read8_at(b, *offset, result)) {
+		return false;
+	}
+	*offset += 1;
+	return true;
+}
+
```

Comment:
for the sake of consistency, can't we just create rz_buf_write/read_ble8_at ? Even just as defines to the rz_buf_write/read8_at functions. It would be less error prone as there's less duplicate code.

Reply:
@ret2libc like this? https://github.com/rizinorg/rizin/pull/2604/commits/88bf4bb41ee35e0ad6c55a92a2f691278abc48c0

Reply:
Why do you need to create this function at all instead of just doing `DEFINE_RZ_BUF_READ_OFFSET_BLE(8)`. I would prefer the latter as it involves less duplication of code.

Reply:
that's what i wrote above "for the sake of consistency, can't we just create rz_buf_write/read_ble8_at ?"